packages/hooks/src/useIsAccountStatusPresent.ts

```diff
@@ -49,11 +49,9 @@ const AccountStatusList = [
 type TAccountStatus = typeof AccountStatusList[number];
 
 export const useIsAccountStatusPresent = (status: TAccountStatus) => {
```

Comment:
This hook doesn't make much sense to me, In my opinion it should be something like this;

```ts

/** A custom hook to check the account status for the current user. */
const useAccountStatus = () => {
    const { data: get_account_status_data, ...rest } = useGetAccountStatus();

    // Add additional information to the account status response.
    const modified_account_status = useMemo(() => {
        if (!get_account_status_data?.status) return;

        return {
            /** Account status. */
            status: get_account_status_data?.status,
            /** client's address is verified by third party services. */
            is_address_verified: get_account_status_data.status.includes('address_verified'),
            /** client is allowed to upload documents. */
            is_allow_document_upload: get_account_status_data.status.includes('allow_document_upload'),
            /** client is age-verified. */
            is_age_verification: get_account_status_data.status.includes('age_verification'),
            /** client is fully authenticated. */
            is_authenticated: get_account_status_data.status.includes('authenticated'),
            /** cashier is locked. */
            is_cashier_locked: get_account_status_data.status.includes('cashier_locked'),
            /** client has updated tax related information. */
            is_crs_tin_information: get_account_status_data.status.includes('crs_tin_information'),
            /** deposit is not allowed. */
            is_deposit_locked: get_account_status_data.status.includes('deposit_locked'),
            /** account is disabled. */
            is_disabled: get_account_status_data.status.includes('disabled'),
            /** client's submitted proof-of-identity documents have expired. */
            is_document_expired: get_account_status_data.status.includes('document_expired'),
            /** client's submitted proof-of-identity documents are expiring within a month. */
            is_document_expiring_soon: get_account_status_data.status.includes('document_expiring_soon'),
            /** Deriv X password is not set. */
            is_dxtrade_password_not_set: get_account_status_data.status.includes('dxtrade_password_not_set'),
            /** client should complete their financial assessment. */
            is_financial_assessment_not_complete: get_account_status_data.status.includes(
                'financial_assessment_not_complete'
            ),
            /** client has not completed financial assessment. */
            is_financial_information_not_complete: get_account_status_data.status.includes(
                'financial_information_not_complete'
            ),
            /** client has accepted financial risk disclosure. */
            is_financial_risk_approval: get_account_status_data.status.includes('financial_risk_approval'),
            /** client has not set financial limits on their account. Applies to UK and Malta clients. */
            is_max_turnover_limit_not_set: get_account_status_data.status.includes('max_turnover_limit_not_set'),
            /** MT5 password is not set. */
            is_mt5_password_not_set: get_account_status_data.status.includes('mt5_password_not_set'),
            /** MT5 deposits allowed, but withdrawal is not allowed. */
            is_mt5_withdrawal_locked: get_account_status_data.status.includes('mt5_withdrawal_locked'),
            /** user must approve the Affiliate's Code of Conduct Agreement. */
            is_needs_affiliate_coc_approval: get_account_status_data.status.includes('needs_affiliate_coc_approval'),
            /** trading is disabled. */
            is_no_trading: get_account_status_data.status.includes('no_trading'),
            /** client cannot trade or withdraw but can deposit. */
            is_no_withdrawal_or_trading: get_account_status_data.status.includes('no_withdrawal_or_trading'),
            /** p2p is blocked for the current payment agent client. */
            is_p2p_blocked_for_pa: get_account_status_data.status.includes('p2p_blocked_for_pa'),
            /** withdrawal through payment agent is allowed. */
            is_pa_withdrawal_explicitly_allowed: get_account_status_data.status.includes(
                'pa_withdrawal_explicitly_allowed'
            ),
            /** this client must reset their password. */
            is_password_reset_required: get_account_status_data.status.includes('password_reset_required'),
            /** this client has opted for a professional account. */
            is_professional: get_account_status_data.status.includes('professional'),
            /** this client has requested for a professional account. */
            is_professional_requested: get_account_status_data.status.includes('professional_requested'),
            /** this client's request for a professional account has been rejected. */
            is_professional_rejected: get_account_status_data.status.includes('professional_rejected'),
            /** this client is using social signup. */
            is_social_signup: get_account_status_data.status.includes('social_signup'),
            /** client has not completed the trading experience questionnaire. */
            is_trading_experience_not_complete: get_account_status_data.status.includes(
                'trading_experience_not_complete'
            ),
            /** client has acknowledged UKGC funds protection notice. */
            is_ukgc_funds_protection: get_account_status_data.status.includes('ukgc_funds_protection'),
            /** client cannot deposit or buy contracts, but can withdraw or sell contracts. */
            is_unwelcome: get_account_status_data.status.includes('unwelcome'),
            /** deposits allowed but withdrawals are not allowed. */
            is_withdrawal_locked: get_account_status_data.status.includes('withdrawal_locked'),
            /** this prevent a client from changing the account currency after deposit attempt. */
            is_deposit_attempt: get_account_status_data.status.includes('deposit_attempt'),
            /** client POI documents name mismatch. */
            is_poi_name_mismatch: get_account_status_data.status.includes('poi_name_mismatch'),
            /** the client can resubmit POA documents. */
            is_allow_poa_resubmission: get_account_status_data.status.includes('allow_poa_resubmission'),
            /** the client can resubmit POI documents. */
            is_allow_poi_resubmission: get_account_status_data.status.includes('allow_poi_resubmission'),
            /** the client has been sharing payment methods. */
            is_shared_payment_method: get_account_status_data.status.includes('shared_payment_method'),
            /** client is not allowed to edit personal profile details. */
            is_personal_details_locked: get_account_status_data.status.includes('personal_details_locked'),
            /** it block any transfer between two accounts. */
            is_transfers_blocked: get_account_status_data.status.includes('transfers_blocked'),
            /** the DF deposit will be blocked until the client gets age verified. */
            is_df_deposit_requires_poi: get_account_status_data.status.includes('df_deposit_requires_poi'),
            /** the client has been fully authenticated by IDV. */
            is_authenticated_with_idv_photoid: get_account_status_data.status.includes(
                'authenticated_with_idv_photoid'
            ),
            /** the client used to be fully authenticated by IDV but it was taken away due to compliance criteria. */
            is_idv_revoked: get_account_status_data.status.includes('idv_revoked'),
        };
    }, [get_account_status_data?.status]);

    return {
        /** The account status response. */
        data: modified_account_status,
        ...rest,
    };
};
```

I would suggest adding the above hook to `@deriv/api` and removing/deprecating `useIsAccountStatusPresent` 🙇🏻

Reply:
I feel this is a lot of loops. May be @yashim-deriv @balakrishna-deriv @matin-deriv  can help with this

Reply:
@likhith-deriv Of course you can improve it buddy 😅 That was just an example

You can convert the array to a `set` and use `has` instead for better performance.

```diff
- const status = get_account_status_data?.status;
+ const status = new Set(get_account_status_data?.status);

- status.includes('disabled');
+ status.has('disabled');
```